### DISCUSSION TITLE
Show dot files

### DISCUSSION THREAD

**Discussion by ciscam** [DirectoryLister/DirectoryLister#701] [Help & Support] [answered]
Hi!

Is it expected behavior that DirectoryLister hides dot files and doesn't include them in .zip-files?

It seems earlier in this project there was a config option 'hide_dot_files', which would add expressions '.\*' and '\*/.\*' to the hidden_files array if TRUE: https://github.com/DirectoryLister/DirectoryLister/issues/92
Adding a line `hide_dot_files' => DI\env('HIDE_DOT_FILES', false),` to app/config/app.php did not change a thing (cache was deleted).
The only file using this variable now (or rather containing 'dot_files') seems to be `app/vendor/symfony/finder/Finder.php`. The object using 'dot_files' in #93, `resources/DirectoryLister.php`, doesn't exist anymore.

Changing the __construct()-Method did enable dotfile display and zip-inclusion (without clearing cache):
```
70c70
<        $this->ignore = static::IGNORE_VCS_FILES | static::IGNORE_DOT_FILES;
---
>        $this->ignore = static::IGNORE_VCS_FILES;
```

Is there a more clean way to do this?

**Comment by PHLAK** [comment, marked as answer]
Thanks for pointing this out. [Symfony Finder](https://symfony.com/doc/current/components/finder.html) is a dependency of this project (not maintained by us) so we can't change that line directly. However, I do think it's a good idea to allow this to be configurable and will look into a way of accomplishing that. Perhaps there's a way to change the behavior of Symfony Finder which would allow users to define a rule to hide dotfiles in the `.hidden` file. Maybe that should even be the default. :thinking:

**Comment by PHLAK** [comment]
FYI: This has been resolved with the [v3.9.0 release](https://github.com/DirectoryLister/DirectoryLister/releases/tag/3.9.0).
